esx_service/vmdk_ops.py

```diff
@@ -1147,44 +1139,19 @@ def getStatusAttached(vmdk_path):
 
     return attached, uuid, attach_as, vm_name
 
-def handle_stale_attach(vmdk_path, kv_uuid):
+def log_attached_volume(vmdk_path, kv_uuid):
        '''
-       Clear volume state for cases where the VM that attached the disk
-       earlier is powered off or removed. Detach the disk from the VM
-       if it's powered off.
+       Log appropriate message for volume thats already attached.
        '''
        cur_vm = findVmByUuid(kv_uuid)
 
        if cur_vm:
-          # Detach the disk only if VM is powered off
-          if cur_vm.runtime.powerState == VM_POWERED_OFF:
-             logging.info("Detaching disk %s from VM(powered off) - %s\n",
-                             vmdk_path, cur_vm.config.name)
-             device = findDeviceByPath(vmdk_path, cur_vm)
-             if device:
-                msg = disk_detach_int(vmdk_path, cur_vm, device)
-                if msg:
-                   msg += " failed to detach disk {0} from VM={1}.".format(vmdk_path,
-                                                                           cur_vm.config.name)
-                   logging.warning(msg)
-                   return err(msg)
-             else:
-                logging.warning("Failed to find disk %s in powered off VM - %s, resetting volume metadata\n",
-                                vmdk_path, cur_vm.config.name)
-                ret = reset_vol_meta(vmdk_path)
-                if ret:
-                   return ret
-          else:
-             msg = "Disk {0} already attached to VM={1}".format(vmdk_path,
-                                                                cur_vm.config.name)
-             logging.warning(msg)
-             return err(msg)
+          msg = "Disk {0} already attached to VM {1}".format(vmdk_path,
+                                                             cur_vm.config.name)
        else:
-          logging.warning("Failed to find VM (id %s) attaching the disk %s, resetting volume metadata",
-                          kv_uuid, vmdk_path)
-          ret = reset_vol_meta(vmdk_path)
-          if ret:
-             return ret
+          msg = "Failed to find VM (id {0}), disk {1} is already attached".format(kv_uuid,
```

Comment:
Agreed on this reasoning. However, renaming the VM is a rare scenario and i think we can still go ahead with printing th VM name we have in the KV.